pwnlib/commandline/libcdb.py

```diff
@@ -136,18 +152,67 @@
 
 common_symbols = ['dup2', 'printf', 'puts', 'read', 'system', 'write']
 
-def find_libc(params):
+
+def find_in_online_mode(params):
     import requests
-    url    = "https://libc.rip/api/find"
+    url = "https://libc.rip/api/find"
     result = requests.post(url, json=params, timeout=20)
     log.debug('Request: %s', params)
     log.debug('Result: %s', result.json())
     if result.status_code != 200 or len(result.json()) == 0:
         log.failure("Could not find libc for %s on libc.rip", params)
         return []
-
     return result.json()
 
+
+def find_in_offline_mode(params):
+    # lookup parser
+    if params.get("symbols"):
+        matching_libcs = libcdb.find_local_libc(params)
+        return matching_libcs if matching_libcs else []
+
+    # hash parser
+    hash_type, hash_value = list(params.items())[0]
+
+    local_db = libcdb._fetch_local_database_path()
+    if not local_db:
+        log.warn_once("The environment variable `PWNLIB_LOCAL_LIBCDB` or `context.local_libcdb` is not configured.")
+        return []
+
+    db_path = Path(local_db) / "db"
+    libs_id = None
+
+    if hash_type == "id":
+        libs_id = hash_value
+    else:
+        if hash_type == "buildid":
+            hash_type = "build_id"
+
+        libc_path = libcdb.search_by_hash(hash_value, hash_type, unstrip=False, offline=True)
+        if libc_path:
+            libs_id = read(libc_path + ".id").decode()
+
+    if libs_id:
+        libc_path = db_path / f"{libs_id}.so"
```

Comment:
Cannot use f-strings, use `"".format()` instead for Python 2 compatibility please.